src/csharp/Microsoft.Spark/Broadcast.cs

```diff
@@ -0,0 +1,160 @@
+        }
+
+        JvmObjectReference IJvmObjectReferenceProvider.Reference => _jvmObject;
+
+        /// <summary>
+        /// Function that creates a file to store the broadcast value in the given path.
+        /// </summary>
+        /// <param name="sc">Spark Context object</param>
+        /// <param name="path">Path where file is to be created</param>
+        /// <param name="value">Broadcast value to be written to the file</param>
+        private void WriteBroadcastValueToFile(SparkContext sc, string path, object value)
+        {
+            Directory.CreateDirectory(sc._temp_dir);
+            FileStream f = File.Create(path);
+            Dump(value, f);
+            f.Close();
+        }
+
+        /// <summary>
+        /// Function that serializes and stores the object passed to the given Stream.
+        /// </summary>
+        /// <param name="value">Serializable object</param>
+        /// <param name="stream">Stream to which the object is serialized</param>
+        private void Dump(object value, Stream stream)
+        {
+            var formatter = new BinaryFormatter();
+            formatter.Serialize(stream, value);
+        }
+
+        /// <summary>
+        /// Get the broadcasted value.
+        /// </summary>
+        /// <returns>The broadcasted value</returns>
+        public object Value()
+        {
+            return BroadcastRegistry.s_registry[_bid];
+        }
+
+        /// <summary>
+        /// Asynchronously delete cached copies of this broadcast on the executors.
+        /// If the broadcast is used after this is called, it will need to be re-sent to each
+        /// executor.
+        /// </summary>
+        public void Unpersist()
+        {
+            _jvmObject.Invoke("unpersist");
+        }
+
+        /// <summary>
+        /// Delete cached copies of this broadcast on the executors. If the broadcast is used after
+        /// this is called, it will need to be re-sent to each executor.
+        /// </summary>
+        /// <param name="blocking">Whether to block until unpersisting has completed</param>
+        public void Unpersist(bool blocking)
+        {
+            _jvmObject.Invoke("unpersist", blocking);
+        }
+
+        /// <summary>
+        /// Destroy all data and metadata related to this broadcast variable. Use this with
+        /// caution; once a broadcast variable has been destroyed, it cannot be used again.
+        /// This method blocks until destroy has completed.
+        /// </summary>
+        public void Destroy()
+        {
+            _jvmObject.Invoke("destroy");
+        }
+    }
+
+    /// <summary>
+    /// Registry to store the broadcast variables for access through workers.
+    /// </summary>
+    internal class BroadcastRegistry
+    {
+        public static ConcurrentDictionary<long, object> s_registry = 
+            new ConcurrentDictionary<long, object>();
+        public static ArrayList s_listBroadcastVariables;
```

Comment:
I'm nervous about these being public fields.  There's nothing stopping someone from going `BroadcastRegistry.s_registry = null` and breaking everything.  What about making them properties with private setters?

Reply:
Makes sense, made them properties with private setters. Thanks!